metagraph/src/cli/transform_anno_tax.cpp

```diff
@@ -0,0 +1,111 @@
+#include "transform_anno_tax.hpp"
+
+#include <filesystem>
+#include <tsl/hopscotch_set.h>
+
+#include "annotation/representation/annotation_matrix/annotation_matrix.hpp"
+#include "annotation/taxonomy/taxonomic_db.hpp"
+#include "common/threads/threading.hpp"
+#include "config/config.hpp"
+#include "cli/load/load_annotation.hpp"
+
+#include "common/logger.hpp"
+
+
+namespace mtg {
+namespace cli {
+
+using mtg::common::logger;
+
+std::vector<std::string> get_anno_filenames(const std::vector<std::string> &paths) {
+    std::vector<std::string> filenames;
+
+    for (const std::string &path: paths) {
+        if (! filesystem::is_directory(path)) {
+            if (!utils::ends_with(path, ".annodbg")) {
+                logger->warn("File '{}' is not an '.annodbg' file. No evaluation for this file.", path);
+                continue;
+            }
+            filenames.push_back(path);
+            continue;
+        }
+        logger->trace("Looking for anno files in directory: '{}'", path);
+        for (auto &rec_file: filesystem::recursive_directory_iterator(path)) {
+            if (utils::ends_with(rec_file.path().string(), ".annodbg")) {
+                logger->trace("Found anno file '{}'", rec_file.path().string());
+                filenames.push_back(rec_file.path().string());
+            }
+        }
+    }
+    return filenames;
+}
+
+void call_taxonomy_map_updates(annot::TaxonomyDB &taxonomy,
+                               const std::vector<std::string> &filenames,
+                               cli::Config *config) {
+    for (uint64_t i = 0; i < filenames.size(); ++i) {
+        const auto &file = filenames[i];
+        std::unique_ptr<annot::MultiLabelEncoded<std::string>> annot =
+                cli::initialize_annotation(file, *config);
+        if (!annot->load(file)) {
+            logger->error("Cannot load annotations from file '{}'", file);
+            exit(1);
+        }
+        taxonomy.kmer_to_taxid_map_update(*annot);
+    }
+}
+
+int transform_anno_tax(Config *config) {
+    assert(config);
+    const std::vector<std::string> &filenames = get_anno_filenames(config->fnames);
+    uint64_t mem_bytes = config->memory_available * 1e9;
+
+    tsl::hopscotch_set<std::string> all_labels;
+
+    // Get all labels from all the annotation files.
+    for (const std::string &file: filenames) {
+        std::unique_ptr<annot::MultiLabelEncoded<std::string>> annot =
+                cli::initialize_annotation(file, *config);
+        if (!annot->load(file)) {
+            logger->error("Cannot load annotations from file '{}'", file);
+            exit(1);
+        }
+        auto file_labels = annot->get_all_labels();
+        for (const std::string &it: file_labels) {
```

Comment:
```suggestion
        for (const std::string &label : annot->get_all_labels()) {
```

Reply:
cool modified everywhere to `elem : set` instead of `elem: set` and will try to avoid empty names like `it`. I let `it` only where iterating pairs where I had to use an object for unzipping the paired elements